portability-core/src/main/java/org/dataportabilityproject/serviceProviders/microsoft/MicrosoftServiceProvider.java

```diff
@@ -44,9 +44,9 @@
         "wl.imap", // outlook export via IMAP
         "wl.offline_access", // provides for refresh tokens
         "wl.calendars", "wl.contacts_calendars"); // calendar export
+
     private final MicrosoftAuth microsoftAuth;
```

Comment:
I am not very familiar with this code, but it's confusing to me that MicrosoftAuth implements OfflineAuthDataGenerator, OnlineAuthDataGenerator, and PasswordAuthDataGenerator also implements OnlineAuthDataGenerator, OfflineAuthDataGenerator. Why do we need both variables?

Reply:
PasswordAuth is the plaintext username/password authenticator that is used for reading calendar via IMAP (I think)
Whereas the MicrosoftAuth is what uses the Oauth flow for microsoft and requests the appropriate scopes (TODO). 

I'm trying to figure out why the Password auth was used for mail, given that I think oauth flow should give us access to mail: 
![image](https://user-images.githubusercontent.com/2763488/35058647-36f6efe8-fb7e-11e7-9261-ec0aa4acf43d.png)

but havent had a chance to debug the flow. @holachuy are you familiar with the reason we're using PasswordAuth for Microsoft mail?

Reply:
ah I remember hearing that MSFT doesn't support OAuth for email. Maybe HTTP API is different than IMAP API. In any case, can you rename the variables consistently (passwordAuth/oauthAuth, mailAuth/calendarAuth etc) or document that one is used for calendar, the other for email?

Reply:
Done, renamed to oauthAuth and passwordAuth since when we add more data types, they may be used for other things.